### PR TITLE
Key value replication using IEntityMessenger #1211

### PR DESCRIPTION
**Pull request by AbfrmBlr**
This only tackles replication. The complete story can be understood from PR #1462.

### CODE REVIEW

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java [hunk 1]
```diff
@@ -327,7 +327,7 @@ private EhcacheEntityResponse invokeInternal(TimeoutDuration timeLimit, EhcacheE
    */
   public static final class Timeouts {
 
-    public static final TimeoutDuration DEFAULT_READ_OPERATION_TIMEOUT = TimeoutDuration.of(5, TimeUnit.SECONDS);
+    public static final TimeoutDuration DEFAULT_READ_OPERATION_TIMEOUT = TimeoutDuration.of(20, TimeUnit.SECONDS);
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java]: Why is this change needed?

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+package org.ehcache.clustered.common.internal.messages;
+
+import org.ehcache.clustered.common.internal.store.Chain;
+
+import java.util.UUID;
+
+/**
+ * This message is sent by the Active Entity to Passive Entity.
+ */
+public class RetirementMessage extends EhcacheEntityMessage {
```
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java]: I do not understand why these messages are named `RetirementMessage`. That's how they are used not what they convey from the active to the passive.

Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java]: ~~For example, something like `PassiveSyncMessage` might express intent better.~~ Scratch that, this one deals specifically with the registration of client ids - let's name it to reflect that!

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  public long getId() {
+    return msgId;
+  }
+
+  public UUID getClientId() {
+    return clientId;
+  }
+
+  public static class ServerStoreRetirementMessage extends RetirementMessage implements ConcurrentEntityMessage {
```
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java]: This is a message that syncs a cluster tier key to the passive - let's name it to reflect that

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java (hunk 1 above)
Comment (AbfrmBlr) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java]: Because the replication tests fail due to get time out. Since the fail-over takes more than 15 secs on my machine atleast.

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java (hunk 3 above)
Comment (AbfrmBlr) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java]: `KeyValueReplicationMessage` ?

Comment (AbfrmBlr) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/RetirementMessage.java]: `ChainReplicationMessage` ?

**Comment by ljacomet** [comment]
Ok with the proposed name changes (that I have 👍 ). Do not forget to update codecs and other related utility classes.

**Comment by AbfrmBlr** [comment]
Review comments addressed

**Comment by AbfrmBlr** [comment]
test this please

**Comment by AbfrmBlr** [comment]
rebased
